src/special-types-and-traits.md

```diff
@@ -0,0 +1,127 @@
+# Special types and traits
+
+Certain types and traits that exist in [the standard library] are known to the
+Rust compiler (usually by [lang items]). This chapter documents the special
+features of these types and traits.
+
+## `Box<T>`
+
+[`Box<T>`] has a few special features that Rust doesn't currently allow for user
+defined types.
+
+* The [dereference operator] for `Box<T>` produces an lvalue which can be moved
+  from. This means that the `*` operator and the destructor of `Box<T>` are
+  built in to the language.
+* [Methods] can take `Box<Self>` as a receiver.
+* For the purpose of the [orphan rules], `Box<T>` is treated as being defined in
+  the same crate as `T` (rather than in `alloc`).
+
+## `UnsafeCell<T>`
+
+[`std::cell::UnsafeCell<T>`] is used for [interior mutability]. It ensures that
+the compiler doesn't perform optimisations that are incorrect for such types.
+It also ensures that [`static` items] which have a type with interior
+mutability aren't placed in memory marked as read only.
+
+## `PhantomData<T>`
+
+[`std::marker::PhantomData<T>`] is a zero-sized, minimum alignment, type that
+is considered to own a `T` for the purposes of [variance] and [drop check].
+
+## Operator Traits
+
+The traits in [`std::ops`] and [`std::cmp`] are used to overload [operators],
+[indexing expressions] and [call expressions].
+
+## `Deref` and `DerefMut`
+
+As well as overloading the unary `*` operator, [`Deref`] and [`DerefMut`] are
+also used in [method resolution] and [deref coercions].
+
+## `Drop`
+
+The [`Drop`] trait provides a [destructor], to be run whenever a value of this
+type is to be destroyed.
+
+## `Copy`
+
+The [`Copy`] trait changes the semantics of a type implementing it. Values
+whose type implements `Copy` are copied rather than moved upon assignment.
+`Copy` cannot be implemented for types which implement `Drop`, or which have
+fields that are not `Copy`. `Copy` is implemented by the compiler for
+
+* [Numeric types]
+* `char` and `bool`
+* [Tuples] of `Copy` types
+* [Arrays] of `Copy` types
+* [Shared references]
+* [Raw pointers]
+* [Function pointers] and [function item types]
+
+## `Clone`
+
+The [`Clone`] trait is implemented by the compiler for the following types:
```

Comment:
Perhaps a note that it's not special, except that it is a bounds on the Copy trait?